### DISCUSSION TITLE
Help needed: Loading the wasm wheel of igraph

### DISCUSSION THREAD

**Discussion by flange-ipb** [pyodide/pyodide#2974] [Q&A] [answered]
Hello!
With the fantastic support of the igraph developers I was able to build a wasm wheel, which Pyodide fails to load at the moment. I'm pretty clueless at the moment. :cry: 

The application is deployed [here](https://flange-ipb.github.io/igraph-wasm-wheel/).
Firefox logs:
> LinkError: imported function 'env.time' signature mismatch pyodide.asm.js:10:135910

and
> Uncaught (in promise) Error: bad export type for `igraphmodule_ARPACKOptionsType`: undefined
    reportUndefinedSymbols https://cdn.jsdelivr.net/pyodide/v0.21.0/full/pyodide.asm.js:10
    loadPackage https://cdn.jsdelivr.net/pyodide/v0.21.0/full/pyodide.asm.js:10
    startPyodide https://flange-ipb.github.io/igraph-wasm-wheel/:13
    async* https://flange-ipb.github.io/igraph-wasm-wheel/:17
pyodide.asm.js:10:256653

You can find my build pipeline [here](https://github.com/igraph/python-igraph/issues/560#issuecomment-1213208122), including a hacky patch to setup.py for building igraph's C code.

**Comment by hoodmane** [comment, marked as answer]
Answered in the `python-igraph` repo: https://github.com/igraph/python-igraph/issues/560#issuecomment-1216447886

> This reveals several interesting issues: 
> 1. We ABI tagged the wheel incorrectly. You built the wheel with Emscripten 3.1.18 so it should be tagged `emscripten_3_1_18` not `emscripten_3_1_14`.
> 2. `packaging.utils.parse_wheel_filename` doesn't recognize the wheel as a properly named wheel. I think this is because of the `7f028d6-231bb65` junk in the name. Where did `7f028d6-231bb65` come from? Was it generated from some tool? In `loadPackage` we use a pared down wheel name parser and accept that we are installing a package called `igraph-develop-7f028d6` for some reason.
> 3. We do not perform the ABI check when loading a wheel with `pyodide.loadPackage` so even if the ABI tag of the wheel were correct we wouldn't error out.

**Reply by flange-ipb** [reply]
Kudos for your quick reply! There should be a warning about the Emscripten version in the [documentation](https://pyodide.org/en/stable/development/new-packages.html#building-python-wheels-out-of-tree).
